library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.format.text
+
+sealed class Type {
+
+    class ValueType {}
+
+    class ResultType {}
+
+    class FunctionType {}
+
+    class Param {}
+
+    class Result {}
+
+    class Limits {}
+
+    class MemoryType {}
+
+    class TableType {}
+
+    class ElementType {}
+
+    class GlobalType {}
```

Comment:
These will all eventually be subclasses of `Type`, but I think it's okay to leave as-is for now.

Reply:
Updated the file, how's it look now?